lib/cretonne/src/verifier.rs

```diff
@@ -614,6 +615,28 @@ impl<'a> Verifier<'a> {
         Ok(())
     }
 
+    fn cfg_integrity(&self, cfg: &ControlFlowGraph) -> Result<()> {
+        for ebb in self.cfg.ebbs() {
+            let cfg_successors = cfg.get_successors(ebb);
+            let cfg_predecessors = cfg.get_predecessors(ebb);
+            for &succ in self.cfg.get_successors(ebb) {
+                if !cfg_successors.contains(&succ) {
```

Comment:
That makes sense to me.

Go ahead and add `Ord` to BasicBlock. I bet you can even `#[derive(PartialOrd, Ord)]` which should give you a lexicographical order.

Alternatively, you can just build a set of the predecessor *instructions* only. The EBB part of `BasicBlock` is actually redundant since you can get it from `layout.inst_ebb()`. (Maybe that's one more thing to verify...)